opendatahub-io/odh-dashboard
# Fixes: Tabs are continuously re-rendered on metrics page (RHOAIENG-589)

Closes: [RHOAIENG-589](https://issues.redhat.com//browse/RHOAIENG-589)

## Description


Fixes a bug where, tabs of the model serving metrics view are continuously re-rendered without any user input.

## How Has This Been Tested?



The fix changes how the "BiasTab" component verifies there is no stale data in the browser session store, as such testing needs to verify a few behaviours.

Test environment setup:
- Setup your cluster to run TrustyAI with some demo data
- Create a number of bias metric  configs for testing, 3 should be sufficient.

Test 1:
- Navigate to Model serving metrics page
- Verify tabs are not re-rendering continuously by opening the "Elements" view of your web console and observing relative calm.

Test 2:
- Navigate to the Model bias tab
- Verify as in Test 1
- Select two charts from the "Metrics to display" dropdown
- Click the configure link in the top right
- Delete one of the configs you selected previously
- Click the "View metrics" button in the top right
- Verify that the deleted chart is not selected in the "Metrics to display" multi-select.

Test 3:
- On the model bias tab, deselect all metrics
- Verify the empty state appears
- Hit the refresh button on your browser
- Verify that one chart is selected and displayed (it is arbitrarily chosen so it doesn't matter which)

## Test Impact


N/A -- bug fix and no tests pre-exist for this area (they will be added in a separate ticket)
 
## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
